README.md

```diff
@@ -53,3 +53,153 @@ The output values are the rank of the low-rank approximation, assembly time, and
+    - 0 : runs the benchmark with regard to the size of the problem,
+    - 1 : runs the benchmark with regard to the number of thread,
+    - 2 : runs the benchmark with regard to the ratio problem size to number of thread.
+- <symmetry_type>: specifies the symmetry type of the matrix. The available symmetry types are:
+    - N : no symmetry (by default value),
+    - S : symmetric matrix.
+- <is_log_scale>: specifies whether to use a logarithmic scale for the x-axis. The available options are :
+    - True : uses a logarithmic scale,
+    - False : uses a linear scale (by default value).
+
+For example, to run the benchmark with test case 1, symmetry type S, execute the following command:
+````bash
+./bench_hmatrix_build 1 S
+````
+
+Note that the benchmark will output the results to a CSV file named <bench_type>_vs_<test_case>.csv in the /build directory. Then, the Python script will create a corresponding directory in /output and copy the CSV file into it. Finally, the graph result will also be saved in this directory.
+
+
+
+### Benchmark customization
+The benchmark parameters are accessible in the hpp files `bench_hmatrix_build`, `bench_hmatrix_matrix_product` and `bench_hmatrix_factorization` located in the `include/htool_benchmark` folder. The user will find the following parameters in the `custom parameters` section of the code : 
+- `number_of_repetitions`: the number of repetitions of the benchmark,
+- `List_algo_type` : the set of implementations,  
+- `List_epsilon` : the set of epsilon values (the tolerance which controls the relative error on block approximation),  
+- `eta`: eligibility constant in the admissibility condition (section 'Hierarchical clustering' [here](https://pmarchand.pages.math.cnrs.fr/htool_documentation/introduction/overview.html)),  
+- `List_pbl_size`: the set of problem size values,  
+- `List_thread`: the set of values for the number of threads,
+- `number_of_products` : the number of Hmatrix products timed,
+- `number_of_solves` : the number of linear system solves timed,
+- `trans` : form of the system of equations A * X = B or A**T * X = B.
+
+The plots of the results can also be customized by modifying the `custom_parameters` function in the `plot_bench.py` file located in the `src` folder. The user will find the following parameters : 
+- `SubList_dim`: the subset of problem size values to display,  
+- `SubList_thread`: the subset of values for the number of threads to display,
+- `SubList_algo_type` : the subset of implementations to display,  
+- `SubList_epsilon` : the subset of epsilon values to display,  
+- `log_exponent` : the exponent 'a' of the logarithm for graph rescaling by N log<sup>a</sup> N in the case of Hmatrix building.
+- `data` : the variable to be plotted. Note that in the factorization benchmark, the user can modify `data` to plot either the factorization time or the solve time.
+
+### Results with htool_benchmark version `TODO`
```

Comment:
I like the idea, but may be it should be in a separate markdown file to avoid cluttering up this file.